src/script/conversation/ConversationRepository.ts

```diff
@@ -1130,7 +1131,8 @@ export class ConversationRepository {
     const localMLSConversation = this.conversationState.findMLS1to1Conversation(otherUserId);
 
     if (protocol === ConversationProtocol.MLS || localMLSConversation) {
-      return this.initMLS1to1Conversation(otherUserId, isMLSSupportedByTheOtherUser);
+      const shouldDelayMLSGroupEstablishment = isLiveUpdate && isMLSSupportedByTheOtherUser;
```

Comment:
When the other user update their supported protocols, we're receiving user.update event that could come before the welcome message for 1:1 conversation with them. We have to add a delay to make sure the welcome message is not wasted, in case the self client would establish mls group before receiving it.

Reply:
This would be a precious "in-code" comment to add  there 😉